workitem/expression_compiler.go

```diff
@@ -36,6 +36,10 @@ func bubbleUpJSONContext(exp criteria.Expression) bool {
 		if t.Left().Annotation(jsonAnnotation) == true || t.Right().Annotation(jsonAnnotation) == true {
 			t.SetAnnotation(jsonAnnotation, true)
 		}
+	case *criteria.SubstringExpression:
```

Comment:
can't you merge these cases(`*criteria.SubstringExpression`,  `*criteria.NotExpression`, etc.) altogether ?

Reply:
Looks like it will not work: https://play.golang.org/p/lPkI5UeSKT

Reply:
ah right. Too bad :(